node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -471,77 +512,75 @@ var UserTotals = function() {
+    };
+    var property = dictionary[activityType];
+
+    // Increment the appropriate property if a translation was found.
+    if (_.has(that, property)) {
+      that[property]++;
+    }
+  };
 
-/**
- * Given an activity type, find the appropriate running-total property and increment by one.
- *
- * @param  {String}    activityType     The activity type to be translated
- */
-UserTotals.prototype.incrementActivities = function(activityType) {
-  // Translate activity type to property name.
-  var property = {
-    'get_asset_comment': 'commentsReceived',
-    'get_asset_comment_reply': 'commentsReceived',    
-    'get_like': 'likesReceived'
-  }[activityType];
-
-  // Increment the appropriate property if a translation was found.  
-  if (_.has(this, property)) {
-    this[property]++;
-  }
+  // Define a method to increment point totals.
+  that.incrementPoints = getPointsIncrementer(that);
+
+  return that;
 };
 
 /**
- * Given an activity type and points, find the appropriate running-total property and add points.
- * This method is used both by CourseTotals and by UserTotals.
+ * Define a method to increment points on a running-totals object.
+ *
+ * @param  {Object}    totals    The running-totals object on which to increment points
+ * @return {Function}            Points-incrementing method
+ * @api private
  *
- * @param  {String}    activityType     The activity type to be translated
- * @param  {Number}    points           The number of points
  */
-CourseTotals.prototype.incrementPoints = UserTotals.prototype.incrementPoints = function(activityType, points) {
-  var property = {
-    // Properties for activity types stored in the database.
-    'add_asset': 'pointsFromAssetsUploaded',
-    'asset_comment': 'pointsFromComments',
-    'export_whiteboard': 'pointsFromAssetsUploaded',
-    'get_asset_comment': 'pointsFromCommentsReceived',
-    'get_asset_comment_reply': 'pointsFromCommentsReceived',
-    'get_like': 'pointsFromLikesReceived',
-    'like': 'pointsFromLikes',
-    // TODO: These activity types for whiteboards are not yet implemented.
-    'whiteboard_add_asset': 'pointsFromWhiteboards',
-    'whiteboard_chat': 'pointsFromWhiteboards',
-    // Generic activity types used for summary totals.
-    'collected': 'pointsCollected',
-    'generated': 'pointsGenerated',
-    'received': 'pointsReceived'
-  }[activityType];
-
-  // Increment the appropriate property by the given number of points if a translation was found.  
-  if (_.has(this, property)) {
-    this[property] += points;
-  }
+var getPointsIncrementer = function(totals) {
+  /**
+   * Given an activity type and points, find the appropriate running-total property and add points.
+   *
+   * @param  {String}    activityType     The activity type to be translated
+   * @param  {Number}    points           The number of points
+   */
+  return function(activityType, points) {
+    // Translate activity type to property name.
+    var dictionary = {
```

Comment:
Similar to the above, I would pass the dictionary into the function creator function rather than hardcode it inside of it